compiler/src/dotty/tools/dotc/transform/LazyVals.scala

```diff
@@ -2,31 +2,39 @@ package dotty.tools.dotc
 package transform
 
 import java.util.IdentityHashMap
-
 import ast.tpd
 import core.Annotations.Annotation
 import core.Constants.Constant
-import core.Contexts._
-import core.Decorators._
+import core.Contexts.*
+import core.Decorators.*
 import core.DenotTransformers.IdentityDenotTransformer
-import core.Flags._
-import core.NameKinds.{LazyBitMapName, LazyLocalInitName, LazyLocalName, ExpandedName}
+import core.Flags.*
+import core.NameKinds.{ExpandedName, LazyBitMapName, LazyLocalInitName, LazyLocalName}
 import core.StdNames.nme
-import core.Symbols._
-import core.Types._
+import core.Symbols.*
+import core.Types.*
 import core.{Names, StdNames}
+import dotty.tools.dotc.config.Feature
 import transform.MegaPhase.MiniPhase
-import transform.SymUtils._
+import transform.SymUtils.*
+
 import scala.collection.mutable
 
 class LazyVals extends MiniPhase with IdentityDenotTransformer {
   import LazyVals._
   import tpd._
 
-  /** this map contains mutable state of transformation: OffsetDefs to be appended to companion object definitions,
-    * and number of bits currently used */
-  class OffsetInfo(var defs: List[Tree], var ord:Int)
+  /**
+   * The map contains the list of the offset trees.
+   */
+  class OffsetInfo(var defs: List[Tree])
+  /** 
+   * This map contains mutable state of transformation: OffsetDefs to be appended 
+   * to companion object definitions, and number of bits currently used. 
+   */
+  class LegacyOffsetInfo(defs: List[Tree], var ord: Int) extends OffsetInfo(defs)
   private val appendOffsetDefs = mutable.Map.empty[Symbol, OffsetInfo]
+  private val oldAppendOffsetDefs = mutable.Map.empty[Symbol, LegacyOffsetInfo]
```

Comment:
Have you considered keeping a unique `OffsetInfo(defs, ord)` structure, and ignoring `ord`(set it to `0` and not read it) in the new codegen? This would avoid these two maps, plus paying attention to select the right one in `transformTemplate`.